### PR TITLE
Make pasted "img" tag has the same behavior as markdown image

### LINKED ISSUE #31230: Markdown image paste regressions
Regression of  Downscale pasted PNG images based on metadata #29123 :

1. The pasted image is not clickable (unable to view the full image)
2. The URL path doesn't contain sub-path if the Gitea instance is deployed under a sub-path.
    * From discord report: Pasting image from the clipboard the Javascript (Paste.js) embeds the image via html code but the src attribute is set to value: `/attachments/<GUID>`. Using a configuration in which the Gitea application is served on a subpath of the domain name, for example: `https://domain.name/git` the image is not rendered.

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #31230

### CODE REVIEW

**Comment by wxiaoguang** [comment]
image

File: modules/markup/html_internal_test.go [hunk 1]
```diff
@@ -18,8 +18,7 @@ import (
 
 const (
 	TestAppURL  = "http://localhost:3000/"
-	TestOrgRepo = "gogits/gogs"
-	TestRepoURL = TestAppURL + TestOrgRepo + "/"
+	TestRepoURL = TestAppURL + "owenr/repo/"
```
Comment (lunny) [inline modules/markup/html_internal_test.go]:
```suggestion
	TestRepoURL = TestAppURL + "owner/repo/"
```

**Comment by silverwind** [comment]
Looks good, I'm surprised that no changes were needed in `modules/markup/markdown/transform_image.go` to avoid double link wrapping as I thought all markup elements go through `html.go` eventually.

**Comment by silverwind** [comment]
Failure related:

```
--- FAIL: TestRender_CrossReferences (0.01s)
    html_test.go:119: 
        	Error Trace:	/home/runner/work/gitea/gitea/modules/markup/html_test.go:119
        	            				/home/runner/work/gitea/gitea/modules/markup/html_test.go:122
        	Error:      	Not equal: 
        	            	expected: "<p><a href=\"http://localhost:3000/gogits/gogs/issues/12345\" class=\"ref-issue\" rel=\"nofollow\">test-owner/test-repo#12345</a></p>"
        	            	actual  : "<p><a href=\"http://localhost:3000/test-owner/test-repo/issues/12345\" class=\"ref-issue\" rel=\"nofollow\">test-owner/test-repo#12345</a></p>"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-<p><a href="http://localhost:3000/gogits/gogs/issues/12345" class="ref-issue" rel="nofollow">test-owner/test-repo#12345</a></p>
        	            	+<p><a href="http://localhost:3000/test-owner/test-repo/issues/12345" class="ref-issue" rel="nofollow">test-owner/test-repo#12345</a></p>
        	Test:       	TestRender_CrossReferences

**Comment by wxiaoguang** [comment]
> Failure related:

Yup, the old names are quite unclear for writing tests, so I decided to rename them by the way.

**Comment by silverwind** [comment]
Ah yes, gogs name need to be removed 😆.

**Review by lunny — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**